ravenframework/Optimizers/fitness/fitness.py

```diff
@@ -139,23 +143,42 @@ def feasibleFirst(rlz,**kwargs):
           'constraintFunction', xr.Dataarray, containing all constraint functions (explicit and implicit) evaluations for the whole population
     @ Out, fitness, xr.DataArray, the fitness function of the given objective corresponding to a specific chromosome.
   """
-  objVar = kwargs['objVar']
-  g = kwargs['constraintFunction']
-  data = np.atleast_1d(rlz[objVar].data)
-  worstObj = max(data)
-  fitness = []
-  for ind in range(data.size):
-    if np.all(g.data[ind, :]>=0):
-      fit=(data[ind])
-    else:
-      fit = worstObj
-      for constInd,_ in enumerate(g['Constraint'].data):
-        fit+=(max(0,-1 * g.data[ind, constInd]))
-    fitness.append(-1 * fit)
-  fitness = xr.DataArray(np.array(fitness),
+  optType = kwargs['type']
+  objVar = [kwargs['objVar']] if isinstance(kwargs['objVar'], str) == True else kwargs['objVar']
+  a = 1.0                     if kwargs['a'] == None                       else kwargs['a']
+  if kwargs['constraintNum'] == 0:
```

Comment:
this needs to be discussed with you. Since 'b' is the list containing multiple weight values, it may need to be treated differently. 

![image](https://github.com/Jimmy-INL/raven/assets/110840088/760e96d3-46a4-426d-a0af-ceccb9aba710)

![image](https://github.com/Jimmy-INL/raven/assets/110840088/fa31eee3-e9db-4bc8-8b78-75146eb6084f)